### ISSUE TITLE
panic in SetModTime

### ISSUE THREAD

**Issue by DaElf** [rclone/rclone#3651]
#### What is the problem you are having with rclone?
I'm hitting this panic often when trying to copy a large number of files from a local EBS vol to s3

```
panic: assignment to entry in nil map

goroutine 40 [running]:
github.com/rclone/rclone/backend/s3.(*Object).SetModTime(0xc002689570, 0x1e08a80, 0xc0000bc3c0, 0x0, 0xed4f3cc54, 0x2acf240, 0x1, 0x1)
        /home/ec2-user/rclone/backend/s3/s3.go:1919 +0x111
github.com/rclone/rclone/fs/operations.equal(0x1e08a80, 0xc0000bc3c0, 0x7f527360f038, 0xc0031ac360, 0x1e176c0, 0xc002689570, 0x10000, 0x7f527360f038)
        /home/ec2-user/rclone/fs/operations/operations.go:221 +0x5e9
github.com/rclone/rclone/fs/operations.Equal(0x1e08a80, 0xc0000bc3c0, 0x7f527360f038, 0xc0031ac360, 0x1e176c0, 0xc002689570, 0xc0060c0090)
        /home/ec2-user/rclone/fs/operations/operations.go:117 +0x88
github.com/rclone/rclone/fs/operations.NeedTransfer(0x1e08a80, 0xc0000bc3c0, 0x1e176c0, 0xc002689570, 0x1e16dc0, 0xc0031ac360, 0x1e16dc0)
        /home/ec2-user/rclone/fs/operations/operations.go:1564 +0x72a
github.com/rclone/rclone/fs/sync.(*syncCopyMove).pairChecker(0xc000554000, 0xc0000bc240, 0xc0000bc280, 0xc0005540c0)
        /home/ec2-user/rclone/fs/sync/sync.go:226 +0x2e3
created by github.com/rclone/rclone/fs/sync.(*syncCopyMove).startCheckers
        /home/ec2-user/rclone/fs/sync/sync.go:307 +0x9a
```

#### What is your rclone version (output from `rclone version`)

Latest code from master branch

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)

Amazon Linux 2

####  Which cloud storage system are you using? (eg Google Drive)

Local file system to aws S3

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)

rclone copy /XXXXX/XXXXX s3:XXXXX/XXXXX/ --progress

**Comment by ncw** [comment]
This looks like that for some reason the SDK returned nil metadata on an object.

Can you try this

https://beta.rclone.org/branch/v1.49.5-245-gcf91aea5-fix-3651-s3-beta/ (uploaded in 15-30 mins)

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.50
